packages/store/src/StoreCore.sol

```diff
@@ -39,6 +40,11 @@ library StoreCore {
     // Register other internal tables
     Hooks.registerSchema();
     StoreMetadata.registerSchema();
+    // Then set their table metadata. If this is done before `registerSchema`,
+    // then downstream validation will fail due to `StoreMetadata` and `Hooks`
+    // loosely depending on one another (by way of `StoreCore` internals).
+    Hooks.setMetadata();
+    StoreMetadata.setMetadata();
```

Comment:
@dk1a felt cleaner to just separate the two paths rather than reproduce their logic here

It might be worth adding an `initialize` function for each of the autogen libs that does the `registerSchema` and `setMetadata` so that downstream usages can use that. It's just here that we have call them separately because of the intertwined dependency.

Reply:
> @dk1a felt cleaner to just separate the two paths rather than reproduce their logic here (I think you suggested this initially anyway)

yeah this does look better than calling `StoreSwitch` directly